product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,510 @@
+- After confirming the target node is not the write leader, stop Postgres.
+- Install the new Postgres version and bdr_pg_upgrade.
+- Initialize the new Postgres version.
+- Copy the configuration for the old Postgres version to the configuration for the new Postgres version.
+- Rename the PGDATA directories.
+- Stop all Postgres instances.
+- Check that you can upgrade safely.
+- Run the Postgres major version upgrade.
+- Modify the Postgres service.
+- Start Postgres.
+- Validate the new Postgres version. 
+- Test the upgrade directly against the newly upgraded node.
+- Cleanup post-upgrade. 
+
+We will look at these steps in more detail below, specifically for upgrading a node from Postgres 15 to Postgres 16. 
+
+## Worked example
+
+A three-node cluster (kaftan, kaolin, and kaboom) we [deployed on AWS using TPA](https://www.enterprisedb.com/docs/pgd/latest/quickstart/quick_start_aws/) is currently running Postgres 15 on a Linux OS (Debian in this case). We want to upgrade a single node to Postgres 16 using `bdr_pg_upgrade`, and then upgrade the rest of the cluster similarly. This walkthrough focuses on the detailed steps for the target node, kaboom. 
+
+!!! Note
+Some steps of this process will involve you running commands as the Postgres owner. We will refer to this user as `postgres` throughout when appropriate. If you are running EDB Postgres Advanced Server, you will need to substitute that user with `enterprisedb` in all relevant commands.
+!!!
+
+### Confirm the current Postgres version
+
+Connect to any node that is still running Postgres 15 and use the following command to connect to PGD's `bdrdb` database:
+
+```bash
+sudo -u postgres psql bdrdb
+```
+Then, when connected to bdrdb, use the following query:
+
+```sql
+SELECT * FROM bdr.group_versions_details;
+```
+
+You should see output similar to this for your cluster:
+
+```
+  node_id   | node_name |        postgres_version        | bdr_version 
+------------+-----------+--------------------------------+-------------
+ 3490219809 | kaftan    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2710197610 | kaboom    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2111777360 | kaolin    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+(3 rows)
+```
+
+Confirm that the Postgres version is the expected version. Then, use the `exit` command to return to the node's command prompt.
+
+### Plan upgrade order
+
+The cluster must be available throughout the process (i.e., a *rolling* upgrade). To achieve continuous cluster availability, we must perform a [planned switchover](#performing-a-planned-switchover) of the write leader node before upgrading that node. We also want to keep the number of planned [switchovers](https://www.enterprisedb.com/docs/pgd/latest/cli/command_ref/pgd_switchover/) to a minimum for availability and efficiency. Accordingly, we will upgrade the current write leader last in our upgrade order, so we only have to perform one planned switchover.
```

Comment:
This is content which should be up top in the overall directions, not buried in the worked example. It applies to all rolling upgrades.

Reply:
Moved it.